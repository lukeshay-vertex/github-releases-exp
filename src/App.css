vertex-viewer {
    height: 100vh;
    width: 100%;
}
