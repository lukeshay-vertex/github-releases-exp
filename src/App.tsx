import React from 'react';

const App: React.FC = (): JSX.Element => <div>Hola</div>;

export default App;
