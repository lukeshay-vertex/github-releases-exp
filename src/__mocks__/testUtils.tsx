import TestRenderer from 'react-test-renderer';

export const findDiv = (
  testInstance: TestRenderer.ReactTestInstance
): TestRenderer.ReactTestInstance =>
  testInstance.find((el) => el.type === 'div');

export const findDivs = (
  testInstance: TestRenderer.ReactTestInstance
): TestRenderer.ReactTestInstance[] =>
  testInstance.findAll((el) => el.type === 'div');
