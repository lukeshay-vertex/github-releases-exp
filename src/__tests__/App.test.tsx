import React from 'react';
import TestRender from 'react-test-renderer';
import App from '../App';
import * as TestUtils from '../__mocks__/testUtils';

describe('App', () => {
  let testRenderer: TestRender.ReactTestRenderer;
  let testInstance: TestRender.ReactTestInstance;

  beforeEach(() => {
    testRenderer = TestRender.create(<App />);
    testInstance = testRenderer.root;
  });

  it('should render a div.', async () => {
    expect(TestUtils.findDiv(testInstance)).toBeDefined();
  });

  it('should match tree snapshot.', async () => {
    expect(testRenderer.toTree()).toMatchSnapshot();
  });

  it('should match JSON snapshot.', async () => {
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
