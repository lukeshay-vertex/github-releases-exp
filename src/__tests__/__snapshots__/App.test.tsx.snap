// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match JSON snapshot. 1`] = `
<div>
  Hola
</div>
`;

exports[`App should match tree snapshot. 1`] = `
Object {
  "instance": null,
  "nodeType": "component",
  "props": Object {},
  "rendered": Object {
    "instance": null,
    "nodeType": "host",
    "props": Object {
      "children": "Hola",
    },
    "rendered": Array [
      "Hola",
    ],
    "type": "div",
  },
  "type": [Function],
}
`;
